app/controllers/RejectedNotificationsController.scala

```diff
@@ -40,15 +40,40 @@ class RejectedNotificationsController @Inject() (
 )(implicit ec: ExecutionContext)
     extends FrontendController(mcc) with I18nSupport with Logging {
 
-  def displayPage(id: String, amendment: Boolean): Action[AnyContent] = (authenticate andThen verifyEmail).async { implicit request =>
+  def displayPage(id: String): Action[AnyContent] = (authenticate andThen verifyEmail).async { implicit request =>
     customsDeclareExportsConnector.findDeclaration(id).flatMap {
       case Some(declaration) =>
         customsDeclareExportsConnector.findNotifications(id).map { notifications =>
-          val maybeMrn = notifications.headOption.map(_.mrn)
-          Ok(rejectedNotificationPage(declaration, maybeMrn, if (amendment) Some(id) else None, getRejectedNotificationErrors(notifications)))
+          val messages = messagesApi.preferred(request).messages
+          val mrn = notifications.headOption.map(_.mrn).getOrElse(messages("rejected.notification.mrn.missing"))
+          Ok(rejectedNotificationPage(declaration, mrn, None, getRejectedNotificationErrors(notifications)))
         }
 
-      case _ => errorHandler.internalError(s"Declaration($id) not found??")
+      case _ => errorHandler.internalError(s"Declaration($id) not found for a rejected submission??")
+    }
+  }
+
+  def displayPageOnUnacceptedAmendment(actionId: String): Action[AnyContent] = (authenticate andThen verifyEmail).async { implicit request =>
+    customsDeclareExportsConnector.findAction(actionId).flatMap {
+      case Some(action) =>
+        action.decId match {
+          case Some(declarationId) =>
+            customsDeclareExportsConnector.findDeclaration(declarationId).flatMap {
+              case Some(declaration) =>
+                customsDeclareExportsConnector.findLatestNotification(actionId).map {
+                  case Some(notification) =>
+                    Ok(rejectedNotificationPage(declaration, notification.mrn, Some(declarationId), notification.errors))
+
+                  case _ => errorHandler.internalServerError(s"Failed|rejected amended Notification not found for Action($actionId)??")
+                }
```

Comment:
Looking at this code trying to for comprehension it, I realised you don't need to call the DB to lookup the latest notification for that actionId as you retrieved the Action instance at the start of this block that contains all the notifications associated with it. That makes whole new BE endpoint redundant no?

Reply:
But we need to retrieve the list of errors, isn't it? And they are in the notification. Am I wrong?

Reply:
Ok, ignore above my bad. The Action only contains a NotificationSummary not the full Notification.